src/Dev/TaskRunner.php

```diff
@@ -42,9 +42,28 @@ protected function init()
         }
     }
 
-    public function index()
+    /**
+     * list of options - can be filtered by adding the `q` request variable (e.g. `/dev/tasks/?q=test` OR `vendor/bin/sake dev/tasks q=test`)
+     * @param  HTTPRequest $index
+     * @return DBHTMLText|string - depends on whether it is `Director::is_cli()` request (returns string) or not (returns DBHTMLText)
+     */
+    public function index($request = null)
     {
+        $baseUrl = Director::absoluteBaseURL();
         $tasks = $this->getTasks();
+        $filter = (string) trim($request->requestVar('q'));
```

Comment:
`trim` already casts to string, so the extra casting is not necessary

Reply:
true!